runtime/runtime/src/verifier.rs

```diff
@@ -144,23 +146,45 @@ pub fn verify_and_charge_transaction(
         )?;
     let transaction = &signed_transaction.transaction;
     let signer_id = &transaction.signer_id;
+    let public_key = &transaction.public_key;
 
     let mut signer = match get_account(state_update, signer_id)? {
         Some(signer) => signer,
         None => {
             return Err(InvalidTxError::SignerDoesNotExist { signer_id: signer_id.clone() }.into());
         }
     };
-    let mut access_key = match get_access_key(state_update, signer_id, &transaction.public_key)? {
+    let mut access_key = match get_access_key(state_update, signer_id, public_key)? {
         Some(access_key) => access_key,
-        None => {
-            return Err(InvalidTxError::InvalidAccessKeyError(
-                InvalidAccessKeyError::AccessKeyNotFound {
-                    account_id: signer_id.clone(),
-                    public_key: transaction.public_key.clone(),
-                },
-            )
-            .into());
+        None => match signer_id.get_account_type() {
+            AccountType::EthImplicitAccount => {
+                if derive_account_id_from_public_key(public_key) == *signer_id {
+                    // TODO What about increasing storage usage for that account, because we added access key?
+                    // TODO Can we assume there can only be one access key added to ETH-implicit account, the one from which the address has been derived?
+                    let mut access_key = AccessKey::full_access();
+                    // TODO Transaction nonce must be greater than access_key nonce, so I used 'transaction.nonce - 1'.
+                    // However, there is a potential for replay attack (test_transaction_hash_collision_for_eth_implicit_account_fail does not pass because of that).
+                    // I would set nonce to (block_height - 1) * near_primitives::account::AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER,
+                    // but block_height is passed here as an Option<BlockHeight>. 
```

Comment:
To discuss.
`test_transaction_hash_collision_for_eth_implicit_account_fail` should fail but currently is not.